### ISSUE TITLE
[META] gtk3-rs is no longer being maintained

### ISSUE THREAD

**Issue by piotrfila** [elkowar/eww#1077]
Hello!

As of March 4 the [gtk crate](https://github.com/gtk-rs/gtk3-rs) eww uses is no longer maintained. I also noticed that [the code for the dbusmenu crate has been removed](https://github.com/ralismark/dbusmenu-rs/commit/2ab95fff9182aff0b092793c61d21fc289794cfa).

Are there any plans to migrate to GTK4? It seems [it has not gone well last time](https://github.com/elkowar/eww/issues/109).

**Comment by elkowar** [comment]
In general I'm not _opposed_ to gtk4, but for now it still seems not really viable due to limitations in wayland support (gtk4 layershell seems to still be _very_ hacky, but at least it's _possible_ now, I gues...

I'll need to look into it again once it actually becomes a necessity..

**Comment by bbb651** [comment]
I think it'll be good to keep either this or #109 open for tracking, also [css variables landed](https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/6540) in gtk4 in addition to other css improvements which is a massive improvement to usability. There's also the option of using a different toolkit altogether but it doesn't seem great at the moment, in addition to breaking pretty much all existing configs, qt is a pain with rust, and for rust toolkits [winit layer shell](https://github.com/rust-windowing/winit/issues/2582) [is stack in limbo](https://github.com/rust-windowing/winit/pull/2832) so the only options that seem viable are the [iced popos fork](https://github.com/pop-os/iced) that has a dedicated sctk layer shell backend, or adding support in another toolkit (perhaps [GPUI](https://www.gpui.rs/) 😉).

**Comment by rudrabhoj** [comment]
@bbb651 Divorcing GPUI from Zed IDE is not difficult at all. I used GPUI for my [nunum calculator](https://github.com/rudrabhoj/numnum) project. To make it work we just needed to properly deal with workspace circular dependencies. It is a really high quality library and IMO makes sense for what eww is doing. It has DOM cum react like syntax. But there would be some breaking changes if we move to GPUI, as valid CSS syntax between current GTK3 implementation and future GPUI will differ somewhat.
